src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp

```diff
@@ -853,42 +862,45 @@ void Hydro_DataReconstruction( const real g_ConVar   [][ CUBE(FLU_NXT) ],
+   if ( LR_Limiter != LR_LIMITER_ATHENA )
    {
-      const int i_cc   = NGhost - 1 + idx_slope%N_SLOPE_PPM;
-      const int j_cc   = NGhost - 1 + idx_slope%N_SLOPE_PPM2/N_SLOPE_PPM;
-      const int k_cc   = NGhost - 1 + idx_slope/N_SLOPE_PPM2;
-      const int idx_cc = IDX321( i_cc, j_cc, k_cc, NIn, NIn );
+      CGPU_LOOP( idx_slope, CUBE(N_SLOPE_PPM) )
+      {
+         const int i_cc   = NGhost - 1 + idx_slope%N_SLOPE_PPM;
+         const int j_cc   = NGhost - 1 + idx_slope%N_SLOPE_PPM2/N_SLOPE_PPM;
+         const int k_cc   = NGhost - 1 + idx_slope/N_SLOPE_PPM2;
+         const int idx_cc = IDX321( i_cc, j_cc, k_cc, NIn, NIn );
 
-//    cc_C/L/R: cell-centered variables of the Central/Left/Right cells
-      real cc_C[NCOMP_LR], cc_L[NCOMP_LR], cc_R[NCOMP_LR], Slope_Limiter[NCOMP_LR];
+//       cc_C/L/R: cell-centered variables of the Central/Left/Right cells
+         real cc_C[NCOMP_LR], cc_L[NCOMP_LR], cc_R[NCOMP_LR], Slope_Limiter[NCOMP_LR];
 
-      for (int v=0; v<NCOMP_LR; v++)   cc_C[v] = g_PriVar[v][idx_cc];
+         for (int v=0; v<NCOMP_LR; v++)   cc_C[v] = g_PriVar[v][idx_cc];
 
-//    loop over different spatial directions
-      for (int d=0; d<3; d++)
-      {
-         const int idx_ccL = idx_cc - didx_cc[d];
-         const int idx_ccR = idx_cc + didx_cc[d];
+//       loop over different spatial directions
+         for (int d=0; d<3; d++)
+         {
+            const int idx_ccL = idx_cc - didx_cc[d];
+            const int idx_ccR = idx_cc + didx_cc[d];
 
-#        if ( defined MHD  &&  defined CHAR_RECONSTRUCTION )
-         MHD_GetEigenSystem( cc_C, EigenVal[d], LEigenVec, REigenVec, EoS, d );
-#        endif
+#           if ( defined MHD  &&  defined CHAR_RECONSTRUCTION )
+            MHD_GetEigenSystem( cc_C, EigenVal[d], LEigenVec, REigenVec, EoS, d );
+#           endif
 
-         for (int v=0; v<NCOMP_LR; v++)
-         {
-            cc_L[v] = g_PriVar[v][idx_ccL];
-            cc_R[v] = g_PriVar[v][idx_ccR];
-         }
+            for (int v=0; v<NCOMP_LR; v++)
+            {
+               cc_L[v] = g_PriVar[v][idx_ccL];
+               cc_R[v] = g_PriVar[v][idx_ccR];
+            }
 
-         Hydro_LimitSlope( cc_L, cc_C, cc_R, LR_Limiter, MinMod_Coeff, d,
-                           LEigenVec, REigenVec, Slope_Limiter, EoS );
+            Hydro_LimitSlope( cc_L, cc_C, cc_R, LR_Limiter, MinMod_Coeff, d,
+                              LEigenVec, REigenVec, Slope_Limiter, EoS );
 
-//       store the results to g_Slope_PPM[]
-         for (int v=0; v<NCOMP_LR; v++)   g_Slope_PPM[d][v][idx_slope] = Slope_Limiter[v];
+//          store the results to g_Slope_PPM[]
+            for (int v=0; v<NCOMP_LR; v++)   g_Slope_PPM[d][v][idx_slope] = Slope_Limiter[v];
 
-      } // for (int d=0; d<3; d++)
-   } // CGPU_LOOP( idx_slope, CUBE(N_SLOPE_PPM) )
+         } // for (int d=0; d<3; d++)
+      } // CGPU_LOOP( idx_slope, CUBE(N_SLOPE_PPM) )
+   } // if ( LR_Limiter != LR_LIMITER_ATHENA )
```

Comment:
I think we can move the following `__syncthreads()` to be inside `if ( LR_Limiter != LR_LIMITER_ATHENA ) { ... }`, right?

Reply:
Yes!